File: src/FreeBilling/Data/Entities/Customer.cs
using System.ComponentModel.DataAnnotations;

namespace FreeBilling.Data.Entities;

public class Customer
{
  public int Id { get; set; }
  [Required]
  [MaxLength(255)]
  public string CompanyName { get; set; } = "";
  public Address? Address { get; set; }
  public string? Contact { get; set; }
  public string? PhoneNumber { get; set; }
}

File: src/FrontEnd/freebilling.ui/src/models/customer.ts
import { type Address, AddressSchema, generateEmptyAddress } from "./Address.js";
import { z } from "zod";

export interface Customer {
  id: number;
  companyName: string;
  address: Address | null;
  contact: string | null;
  phoneNumber: string | null;
}

export const CustomerSchema = z.object({
  id: z.number(),
  companyName: z.string().min(1, "Required"),
  address: z.nullable(AddressSchema),
  contact: z.nullable(z.string()),
  phoneNumber: z.nullable(z.string())
});

export function generateEmptyCustomer() {
  return {
    id: 0,
    companyName: "",
    contact: "",
    phoneNumber: "",
    address: generateEmptyAddress()
  } as Customer;
}

export type CustomerErrors = z.inferFormattedError<typeof CustomerSchema>;